discovery/openstack/instance.go

```diff
@@ -113,8 +114,18 @@ func (i *InstanceDiscovery) refresh() (*targetgroup.Group, error) {
 	if err != nil {
 		return nil, fmt.Errorf("could not create OpenStack session: %s", err)
 	}
+
+	availability := gophercloud.AvailabilityPublic
+	switch i.endpointType {
+	case "internal":
+		availability = gophercloud.AvailabilityInternal
+	case "admin":
+		availability = gophercloud.AvailabilityAdmin
+
```

Comment:
This should error if there's an invalid value, preferably at config load time.

Reply:
I added a check for valid values in 1a24f529a6a1bb3a894cca9ba437bb0d04f787ce

Reply:
you can remove this check now that it is enforced when unmarshaling the configuration.